packages/@react-aria/checkbox/src/useCheckbox.ts

```diff
@@ -61,8 +63,22 @@ export function useCheckbox(props: AriaCheckboxProps, state: ToggleState, inputR
     }
   });
 
+  // Reset validation state on label press for checkbox with a hidden input.
+  let {pressProps} = usePress({
+    onPress() {
```

Comment:
```suggestion
  let {pressProps} = usePress({
    isDisabled: isDisabled || isReadOnly,
    onPress() {
```
Minor nitpick, but this should also have its onPress logic disabled when the checkbox is disabled otherwise form validation will trigger when clicking on a disabled checkbox's label